plugin_runner/plugin_runner.py

```diff
@@ -237,17 +240,27 @@ def refresh_event_type_map() -> None:
                 log.warning(f"Unknown RESPONDS_TO type: {type(responds_to)}")
 
 
-def load_plugins() -> None:
+def load_plugins(specified_plugin_paths: list[str] | None = None) -> None:
     """Load the plugins."""
     # first mark each plugin as inactive since we want to remove it from
     # LOADED_PLUGINS if it no longer exists on disk
     for plugin in LOADED_PLUGINS.values():
         plugin["active"] = False
 
-    candidates = os.listdir(PLUGIN_DIRECTORY)
+    if specified_plugin_paths is not None:
+        # convert to Paths
+        plugin_paths = [pathlib.Path(name) for name in specified_plugin_paths]
+
+        for plugin_path in plugin_paths:
+            # when we import plugins we'll use the module name directly so we need to add the plugin
+            # directory to the path
+            path_to_append = f"./{plugin_path.parent}"
+            sys.path.append(path_to_append)
+    else:
+        candidates = os.listdir(PLUGIN_DIRECTORY)
 
-    # convert to Paths
-    plugin_paths = [pathlib.Path(os.path.join(PLUGIN_DIRECTORY, name)) for name in candidates]
+        # convert to Paths
+        plugin_paths = [pathlib.Path(os.path.join(PLUGIN_DIRECTORY, name)) for name in candidates]
```

Comment:
This logic branch preserves previous behavior when running `python plugin_runner/plugin_runner.py`